californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -679,17 +679,20 @@ private void reject(final RawData raw, final MessageFormatException cause) {
 			rst.setToken(new byte[0]);
 			rst.setDestination(raw.getAddress());
 			rst.setDestinationPort(raw.getPort());
-			for (MessageInterceptor interceptor : interceptors) {
-				interceptor.sendEmptyMessage(rst);
-			}
-			connector.send(serializer.serializeEmptyMessage(rst));
+
+			reject(rst);
```

Comment:
ClientSynchronousTest

	@Test
	public void testSignal() throws Exception {
		/* ver. := 01, type:= CON (0), toklen := 0 | code := 7* | MID := 0x0102 */
		byte[] signal = new byte[] { 0x40, 0x07, 0x01, 0x02 };

		try (DatagramSocket socket = new DatagramSocket()) {
			DatagramPacket datagram = new DatagramPacket(signal, signal.length, InetAddress.getLoopbackAddress(), serverPort);
			DatagramPacket receive = new DatagramPacket(new byte[128], 128);
			socket.setSoTimeout(2000);
			socket.send(datagram);
			try {
				socket.receive(receive);
			} catch (SocketTimeoutException e) {
				Assert.fail("timeout instead of RST");
			}
		}
	}

with that test, and a temporary fix in 

	DataParser

	public final Message parseMessage(final RawData raw) {

		DatagramReader reader = new DatagramReader(raw.getBytes());
		MessageHeader header = parseHeader(reader);
		try {
			if (CoAP.isRequest(header.getCode())) {
				return parseMessage(reader, header, new Request(CoAP.Code.valueOf(header.getCode())));
			} else if (CoAP.isResponse(header.getCode())) {
				return parseMessage(reader, header, new Response(CoAP.ResponseCode.valueOf(header.getCode())));
			} else if (CoAP.isEmptyMessage(header.getCode())) {
				return parseMessage(reader, header, new EmptyMessage(header.getType()));
			}
		} catch (MessageFormatException e) {
			throw new MessageFormatException(e.getMessage(), header.getMID(), header.getCode(), CoAP.Type.CON == header.getType());
		}
		throw new MessageFormatException("illegal message code", header.getMID(), header.getCode(), CoAP.Type.CON == header.getType());
	}

the debugger shows the damage:

> reject(final RawData raw, final MessageFormatException cause)

creates a EmpytMessage only with destination and without source.

> private void reject(final Message message)

then creates in `EmptyMessage.newRST(message)` a RST for that 1. Level RST using 

	public static EmptyMessage newRST(Message message) {
		EmptyMessage rst = new EmptyMessage(Type.RST);
		rst.setDestination(message.getSource());
		rst.setDestinationPort(message.getSourcePort());
		rst.setMID(message.getMID());
		return rst;
	}

the empty source address as destination.

Result:
> Exception in network stage thread [UDP-Sender-localhost/127.0.0.1:0[0]]:
> java.net.BindException: Cannot assign requested address: Datagram send failed

(destination port 0)